packages/core/src/components/essentials/tooltip-bar.ts

```diff
@@ -0,0 +1,187 @@
+				}
+				// Fade in
+				this.tooltip.classed("hidden", false);
+			}
+		});
+
+		// listen to hide-tooltip Custom Events to hide the tooltip
+		this.services.events.getDocumentFragment().addEventListener("hide-tooltip", e => {
+			this.tooltip.classed("hidden", true);
+		});
+	}
+
+	/**
+	 * Get the position of the tooltip relative to the active hovered bar. Tooltip should appear above
+	 * positive valued data and below negative value data.
+	 * @param hoveredElement
+	 */
+	getTooltipPosition(hoveredElement) {
+		const data = select(hoveredElement).datum() as any;
+
+		const holderPosition = select(this.services.domUtils.getHolder()).node().getBoundingClientRect();
+		const barPosition = hoveredElement.getBoundingClientRect();
+
+		const { verticalOffset } = this.model.getOptions().tooltip.datapoint;
+		// if there is a negative value bar chart, need to place the tooltip below the bar
+		if (data.value <= 0) {
+			// negative bars
+			const tooltipPos = {
+				left: (barPosition.left - holderPosition.left) + barPosition.width / 2,
+				top: (barPosition.bottom - holderPosition.top) + verticalOffset };
+
+			return {placement: TooltipPosition.BOTTOM, position: tooltipPos};
+		} else {
+			// positive bars
+			const tooltipPos = {
+				left: (barPosition.left - holderPosition.left) + barPosition.width / 2,
+				top: (barPosition.top - holderPosition.top) - verticalOffset };
+
+			return {placement: TooltipPosition.TOP, position: tooltipPos};
+		}
+	}
+
+	/**
+	 * Returns the html for the bar single point tooltip
+	 * @param data associated values for the hovered bar
+	 */
+	getTooltipHTML(data: any) {
+		const formattedValue = Tools.getProperty(this.model.getOptions(), "tooltip", "valueFormatter") ?
+		this.model.getOptions().tooltip.valueFormatter(data.value) : data.value.toLocaleString("en");
+
+		return `<div class="datapoint-tooltip"><p class="value">${formattedValue}</p></div>`;
+	}
+
+	/**
+	 * Multip tooltips for bar charts include totals for each stack
+	 * @param data
+	 */
+	getMultiTooltipHTML(data: any) {
+		const points = data;
+		let total = 0;
+
+		points.reverse();
+
+		// get the total for the tooltip
+		points.forEach(item => total += item.value);
+
+		let listHTML = "<ul class='multi-tooltip'>";
+
+		points.forEach(datapoint => {
+			const formattedValue = Tools.getProperty(this.model.getOptions(), "tooltip", "valueFormatter") ?
+			this.model.getOptions().tooltip.valueFormatter(datapoint.value) : datapoint.value.toLocaleString("en");
+
+			const indicatorColor = this.model.getStrokeColor(datapoint.datasetLabel, datapoint.label, datapoint.value);
+
+			listHTML += `<li><div class="datapoint-tooltip">
+				<a style="background-color:${indicatorColor}" class="tooltip-color"></a>
+				<p class="label">${datapoint.datasetLabel}</p>
+				<p class="value">${formattedValue}</p>
+				</div></li>`;
+		});
+
+		return listHTML + `<li><div class='total-val'><p class='label'>Total</p><p class='value'>${total}</p></div></li></ul>` ;
```

Comment:
Could some of this be simplified a bit ... this return could be formatted a bit nicer, and `listHTML` could be eliminated

Reply:
i'll update both to not to remove listHTML like below



```

		return "<ul class='multi-tooltip'>" +
			points.forEach(datapoint => {
				const formattedValue = Tools.getProperty(this.model.getOptions(), "tooltip", "valueFormatter") ?
				this.model.getOptions().tooltip.valueFormatter(datapoint.value) : datapoint.value.toLocaleString("en");

				const indicatorColor = this.model.getStrokeColor(datapoint.datasetLabel, datapoint.label, datapoint.value);

				return `<li><div class="datapoint-tooltip">
					<a style="background-color:${indicatorColor}" class="tooltip-color"></a>
					<p class="label">${datapoint.datasetLabel}</p>
					<p class="value">${formattedValue}</p>
					</div></li>`;
			}) + "</ul>" ;

```

Reply:
actually more like this 

```

		return  "<ul class='multi-tooltip'>" +
			points.map(datapoint => {
				const formattedValue = Tools.getProperty(this.model.getOptions(), "tooltip", "valueFormatter") ?
				this.model.getOptions().tooltip.valueFormatter(datapoint.value) : datapoint.value.toLocaleString("en");

				const indicatorColor = this.model.getStrokeColor(datapoint.datasetLabel, datapoint.label, datapoint.value);

				return `<li><div class="datapoint-tooltip">
					<a style="background-color:${indicatorColor}" class="tooltip-color"></a>
					<p class="label">${datapoint.datasetLabel}</p>
					<p class="value">${formattedValue}</p>
					</div></li>`;
			}).join("") + `<li><div class='total-val'><p class='label'>Total</p><p class='value'>${total}</p></div></li></ul>`;
```

Reply:
@natashadecoste is this good to go? can we resolve?